uaclient/files.py

```diff
@@ -0,0 +1,344 @@
+import copy
+import json
+import logging
+import os
+from datetime import datetime
+from typing import Dict, Optional
+
+from uaclient import (
+    defaults,
+    event_logger,
+    exceptions,
+    messages,
+    schemas,
+    util,
+)
+from uaclient.machine_token_data import PublicMachineTokenData
+
+event = event_logger.get_event_logger()
+LOG = logging.getLogger(__name__)
+
+
+class UAFile:
+    def __init__(
+        self,
+        name: str,
+        root_directory: str = defaults.DEFAULT_DATA_DIR,
+        private: bool = True,
+    ):
+        self._root_directory = root_directory
+        self._file_name = name
+        self._is_private = private
+        self._path = None
+
+    @property
+    def path(self) -> str:
+        if not self._path:
+            self._path = os.path.join(
+                self._root_directory, self._file_name
+            )  # type: ignore
+        return self._path  # type: ignore
```

Comment:
Does this need to be a property? or can it just be
```suggestion
        self.path =  os.path.join(
            self._root_directory, self._file_name
        )  # type: ignore
```

Reply:
Or if we keep it as a property for encapsulation, we can still pre-compute the value
```suggestion
        self._path = os.path.join(
            self._root_directory, self._file_name
        )  # type: ignore

    @property
    def path(self) -> str:
        return self._path
```

Reply:
Also why are we ignoring types here?

Reply:
I was going ```Incompatible types in assignment (expression has type "str", variable has type "None")``` but with this refactor it solves it.